admin-auth/src/test/kotlin/com/epam/drill/admin/auth/CoffeineCacheServiceTest.kt

```diff
@@ -3,46 +3,55 @@ package com.epam.drill.admin.auth
 import com.epam.drill.admin.auth.principal.Role
 import com.epam.drill.admin.auth.principal.User
 import com.epam.drill.admin.auth.service.impl.CoffeineCacheService
-import com.github.benmanes.caffeine.cache.Caffeine
+import com.github.benmanes.caffeine.cache.Cache
+import io.ktor.auth.*
 import kotlinx.coroutines.runBlocking
-import java.time.Duration
+import org.junit.jupiter.api.AfterEach
+import org.mockito.Mockito
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.times
 import kotlin.test.Test
-import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.fail
+
 
 /**
  * Tests for [CoffeineCacheService] logic
  */
 class CoffeineCacheServiceTest {
 
-    @Test
-    fun `given two distinct apiKeys, then cache size should be two`(): Unit = runBlocking {
-        val cacheService = CoffeineCacheService(
-            Caffeine.newBuilder()
-                .maximumSize(5)
-                .expireAfterWrite(Duration.ofMinutes(1))
-                .build()
-        )
-
-        cacheService.getFromCacheOrPutIfAbsent("custom-key-1") { User(1, "test1", Role.USER) }
-        cacheService.getFromCacheOrPutIfAbsent("custom-key-2") { User(2, "test2", Role.USER) }
-
-        assertEquals(2, cacheService.cache.estimatedSize())
+    private val caffeineMock = mock<Cache<String, Principal>>()
+    private var cacheService = CoffeineCacheService(caffeineMock)
+
+    @AfterEach
+    fun reset() {
+        Mockito.reset(caffeineMock)
     }
 
     @Test
-    fun `given only one apiKey two times, then cache size should be one`(): Unit = runBlocking {
-        val user = User(1, "test1", Role.USER)
-        val cacheService = CoffeineCacheService(
-            Caffeine.newBuilder()
-                .maximumSize(5)
-                .expireAfterWrite(Duration.ofMinutes(1))
-                .build()
-        )
-
-        cacheService.getFromCacheOrPutIfAbsent("custom-key-1") { user }
-        cacheService.getFromCacheOrPutIfAbsent("custom-key-1") { user }
-
-        assertEquals(1, cacheService.cache.estimatedSize())
+    fun `given apiKey, then should be called put method`(): Unit = runBlocking {
+        val key1 = "custom-key-1"
+        val user1 = User(1, "test1", Role.USER)
```

Comment:
Better to rename user1 -> user. Any way `user2` is not exists

Reply:
renamed